pkg/types/spec/validations.go

```diff
@@ -519,21 +561,30 @@ func ExtractAPIConfigs(configBytes []byte, provider types.ProviderType, configFi
 			return nil, errors.Append(err, fmt.Sprintf("\n\napi configuration schema can be found here: https://docs.cortex.dev/v/%s/deployments/api-configuration", consts.CortexVersionMinor))
 		}
 
-		errs = cr.Struct(&api, data, apiValidation(provider, resourceStruct.Kind))
-
+		errs = cr.Struct(&api, data, apiValidation(provider, resourceStruct))
 		if errors.HasError(errs) {
 			name, _ := data[userconfig.NameKey].(string)
 			kindString, _ := data[userconfig.KindKey].(string)
 			kind := userconfig.KindFromString(kindString)
 			err = errors.Wrap(errors.FirstError(errs...), userconfig.IdentifyAPI(configFileName, name, kind, i))
 			return nil, errors.Append(err, fmt.Sprintf("\n\napi configuration schema can be found here: https://docs.cortex.dev/v/%s/deployments/api-configuration", consts.CortexVersionMinor))
 		}
-		api.Index = i
-		api.FileName = configFileName
 
-		api.ApplyDefaultDockerPaths()
+		if resourceStruct.Kind == userconfig.APISplitterKind {
+			if provider == types.LocalProviderType {
+				return nil, ErrorAPISplitterNotSupported(api)
```

Comment:
`return nil, ErrorAPISplitterNotSupported(api)` -> `return nil, errors.Wrap(ErrorAPISplitterNotSupported(), api.Identify())`

and then don't accept api in ErrorAPISplitterNotSupported().

api.Identify() is a bit more useful than printing out api.Name because it also provides the location of yaml file, the index of the api and the name.